spec/components/trainees/record_actions/view_preview.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "govuk/components"
+
+module Trainees
+  module RecordActions
+    class ViewPreview < ViewComponent::Preview
+      def trn_requested_or_received
+        trainee.state = %i[submitted_for_trn trn_received].sample
+        render Trainees::RecordActions::View.new(trainee)
+      end
+
+      def deferred
+        trainee.state = :deferred
```

Comment:
You can probably use the factory now

Reply:
Should we though? All the other preview components have so far avoid using a factory.

Reply:
🤔 let's leave that for the time being